modules/packages/LinearAlgebra.chpl

```diff
@@ -927,6 +927,28 @@ proc trace(A: [?D] ?eltType) {
   return trace;
 }
 
+/* Return the Kronecker Product of matrix ``A`` and matrix ``B``.
+   If the size of A is ``x * y`` and of B is ``a * b`` then size of the resulting
+   matrix will be ``(x * a) * (y * b)`` */
+proc kron(A: [?ADom] ?eltType, B: [?BDom] eltType) {
+  if ADom.rank != 2 || BDom.rank != 2 then compilerError("Rank size not 2");
+  const (rowA, colA) = A.shape;
+  const (rowB, colB) = B.shape;
+  var C = Matrix(rowA*rowB, colA*colB, eltType=eltType);
+  forall i in 0..#rowA {
+    forall j in 0..#colA {
+      var stR = i*rowB;
+      var stC = j*colB;
+      forall k in 0..rowB-1 {
```

Comment:
I would change these 2 inner `forall` loops back into `for` loops. The `forall` loops will only spawn tasks based on the available resources, which will be consumed by the outer `forall` loop tasks.

Reply:
OK, I understand it